modules/juce_core/text/juce_StringPool.cpp

```diff
@@ -161,4 +161,113 @@ StringPool& StringPool::getGlobalPool() noexcept
+
+    // Ensure the input array is sorted and contains no duplicates
+    jassert (std::is_sorted (stringsToAdd.begin(), stringsToAdd.end(), [](const auto& a, const auto& b) {
+        return compareStrings (a, b) < 0;
+    }));
+
+    jassert(std::adjacent_find (stringsToAdd.begin(),
+                               stringsToAdd.end(),
+                               [](const auto& a, const auto& b) { return compareStrings (a, b) == 0; })
+            == stringsToAdd.end());
+
+    // Ensure the existing pool is sorted (very long check)
+    //    jassert(std::is_sorted(
+    //        strings.begin(), strings.end(), [](const auto& a, const auto& b) { return compareStrings(a, b) < 0; }));
+
+    const ScopedLock sl (lock);
+
+    Array<Identifier> result;
+    result.resize (stringsToAdd.size());
+
+    int start = 0;
+    const int end = stringsToAdd.size();
+
+    while (start < end) {
+        const auto& startString = stringsToAdd.getReference (start);
+        auto [found, insertionIndex] = locateOrGetInsertIndex (startString, 0, strings.size());
+
+        if (found) {
+            result.set (start, Identifier (strings.getReference (insertionIndex), true));
+            start++;
+            continue;
+        }
+
+        int numElems;
+        auto low = start + 1;
+        auto high = end;
+
+        auto [endFound, endInsertionIndex] = locateOrGetInsertIndex (stringsToAdd.getLast(), insertionIndex, strings.size());
+
+        // Check if all the remaining elements can be inserted at insertion_index
+        if (!endFound && endInsertionIndex == insertionIndex) {
+
+            numElems = end - start;
+
+        } else {
+            // Binary search to find the last element that can be inserted at insertion_index
+            while (low < high) {
+                auto mid = low + (high - low) / 2;
+                auto [newFound, newInsertionIndex] =
+                    locateOrGetInsertIndex (stringsToAdd.getReference (mid), insertionIndex, strings.size());
+
+                if (newFound || newInsertionIndex != insertionIndex)
+                    high = mid;
+                else
+                    low = mid + 1;
+            }
+
+            numElems = low - start;
+        }
+
+        // Insert the elements
+        strings.insertArray (insertionIndex, stringsToAdd.begin() + start, numElems);
+
+        for (int i = 0; i < numElems; i++)
+            result.set (i + start, Identifier (strings.getReference (insertionIndex + i), true));
+
+        start += numElems;
+    }
+
+    // Ensure the pool is still sorted and contains no duplicates (very long checks)
+    //    jassert(std::is_sorted(
+    //        strings.begin(), strings.end(), [](const auto& a, const auto& b) { return compareStrings(a, b) < 0; }));
+    //
+    //    jassert(std::adjacent_find(
+    //                strings.begin(), strings.end(), [](const auto& a, const auto& b) { return compareStrings(a, b) == 0; })
+    //            == strings.end());

```

Comment:
I wonder if there's a way we could add unit tests to our JUCE modifications like these